InterfaceParser/README.md

```diff
@@ -0,0 +1,114 @@
+# `MOBILE_RPC.xml` structure
```

Comment:
In this README would it be possible to include further information for the different parsers? I am Not really sure what each of these parsers are for (json_rpc.py, sdl_rpc_v1.py, sdl_rpc_v2.py).

Reply:
Can we remove the sdl_rpc_v1.py parser?

Reply:
From our side we also don't know much about the parser types. From SDL core build log we can see all types of parsers is used in different generator calls for different XML types. I will update README with this information. 

`Generating interface source code with following parameters:
    Source xml      : /home/user/sdl/sdl_core/src/components/interfaces/MOBILE_API.xml
    Namespace       : mobile_apis
    Output directory: /home/user/sdl/sdl_core/build/src/components/interfaces
    Parser type     : sdlrpcv2
`

`Generating interface source code with following parameters:
    Source xml      : /home/user/sdl/sdl_core/src/components/interfaces/HMI_API.xml
    Namespace       : hmi_apis
    Output directory: /home/user/sdl/sdl_core/build/src/components/interfaces
    Parser type     : jsonrpc
`

`Generating interface source code with following parameters:
    Source xml      : /home/user/sdl/sdl_core/src/components/interfaces/HMI_API.xml
    Namespace       : rpc::policy_table_interface_base
    Output directory: /home/user/sdl/sdl_core/build/src/components/policy/policy_regular
    Parser type     : hmi-policy-types
`

`Generating interface source code with following parameters:
    Source xml      : /home/user/sdl/sdl_core/src/components/interfaces/MOBILE_API.xml
    Namespace       : rpc::policy_table_interface_base
    Output directory: /home/user/sdl/sdl_core/build/src/components/policy/policy_regular
    Parser type     : mobile-policy-types
`

`Generating interface source code with following parameters:
    Source xml      : /home/user/sdl/sdl_core/src/components/interfaces/v4_protocol_v1_2_no_extra.xml
    Namespace       : ns_smart_device_link_rpc::V1
    Output directory: /home/user/sdl/sdl_core/build/src/components/interfaces
    Parser type     : sdlrpcv1
`

Reply:
Ok. Also I think some short descriptions would be

hmi_api_rpc.py: Parser used during the generation of SDL Core's HMI API classes.

sdl_rpc_v2.py: Parser used during the generation of the RPC Spec Classes used by all project generators.

sdl_rpc_v1.py: Parser used during the generation of SDL Core's Legacy API classes.

Reply:
I'm confused to why some of these are even needed and included here and their names.  `sdl_rpc_v1 ` should really be called something like `sdl_rpc_legacy`, since that's its purpose. The v2 is actually a the first version for the purpose of the proposal. Also why is the `hmi_api_rpc.py` file in this PR? That is _only_ associated with the Core project and is mostly out of the scope of the RPC spec.

Reply:
I think the benefit of including all of the parsers in this project is that it would make the generation process for core easier when it does the submodule import? I haven't seen the core implementation for using this repo as a submodule so I cant say for sure where its best to include the extra parsers.

Also sdl_rpc_v1 and sdl_rpc_v2 have existed in Core for a while. The sdl_rpc_v2 parser wasnt created for this proposal. When I say "legacy" for sdl_rpc_v1 I mean legacy from wayyy back, as in Sync 1 old.

Reply:
This line from the proposal implies that only the Mobile api parser should be included in this repo. So maybe the hmi api parser should be removed:

> The parser code that parses the MOBILE_API should be located in the rpc_spec repository. It will be included into the library repositories due to the submodule reference. This should help maintaining the parser code.

Reply:
@JackLivio @joeygrover I raised [the same question](https://github.com/smartdevicelink/sdl_evolution/issues/853#issuecomment-550334703) on the proposal discussion, but this question was ignored. Then after, the proposal was accepted and the Core was included as the impacted platform. It is clear that because the proposal does not require any deep old code refactoring except the submodule and Python 3 changes, the same time this requires separated place of generator and parser parts, thus obviously the way without breaking changes is to only eject parser from the old code as-is with all supported formats. We also prefer to include only sdl_rpc_v2 as only this is required for MOBILE_API, and we did follow that before the Core was impacted with new revision changes. So please consider the best route and let us know, because in case if we need to do only with sdl_rpc_v2 here, we need to discuss internaly the additional effort for this. And it seems this will block the JS generator release too. @TinaKleczka @kshala-ford @a-prykho please assist.

Reply:
Done. This repository contains the base code of parser and now supports only the `sdlrpcv2` version required for MOBILE_API. All other versions were returned to `sdl_core`. The same time `sdl_core` generator continues to use this `rpc_spec` repository as the submodule, as requested in the proposal.

Reply:
Thank you @vladmu. I am sorry we missed your question in the proposal discussion. I think keeping the "core only" specific generators in the core repository is a good solution. 

Are there any outstanding questions relating to the Core implementation of this feature? If we  understand each other, Core will continue to use the original parsers for the sdlrpcv1 and json generators. Core will be updated to use the parser in this repository as a submodule for generating the mobile api.